Runtime/Scripts/DataCollection/RGDataCollection.cs

```diff
@@ -106,26 +108,60 @@ public void SaveReplayDataInfo(long clientId, RGStateActionReplayData replayData
             RGDebug.LogVerbose($"DataCollection[{clientId}] - Saving replay data for tick {replayData.tickInfo.tick}");
             
             // Add the new replay data to a new or existing mapping in our client replay data dictionary
-            var replayDatas = _clientIdToReplayData.GetOrAdd(clientId, new List<RGStateActionReplayData>());
-            replayDatas.Add(replayData);
+            var replayDataTasks = _clientIdToReplayDataTasks.GetOrAdd(clientId, v => new List<Task>());
+            var validationDataTasks = _clientIdToValidationDataTasks.GetOrAdd(clientId, v => new List<Task>());
```

Comment:
Are we guaranteed that this isn't being called simultaneously for the _same `clientId` value_? The `List<Task>` you're working with isn't thread-safe, but that should be alright as long as we're only processing one `SaveReplayDataInfo` for each `clientId` at a time.

Reply:
Good callout, but yes we are guaranteed not to call this in parallel for the same clientId.  This is called from each client botrunner thread in a manner that blocks said thread until this completes.